docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,197 @@
+
+-   ***S3BlockUploadOperator***
+    -   **S3BlockUpload** operator upload the blocks into S3 using different API's which depends on number of blocks of a file.
+If the file has single block then upload the block into S3 using **AmazonS3Client.putObject(...)** API. S3 returns a response of type **PutObjectResult** includes the **ETag**.
+If the file has more blocks then upload the block using **AmazonS3Client.uploadPart(...)** method. S3 returns a response of type UploadPartResult includes the **ETag**. This **ETag** value must be include in request to complete multipart upload.
+    -   **S3BlockUpload** emits **(path, ETag)** to **s3FileMerger**.
+
+-   ***S3FileMerger***
+    -   Complete multipart upload request using **AmazonS3Client.completeMultipartUpload(...)** and must include the **upload ID**, a list of both part numbers and corresponding **ETag** values. **S3FileMergerOperator** send the complete multi-part upload request to S3 once it have all the **part ETag's** of a file. 
+**Amazon S3** creates an object by concatenating the parts in ascending order based on part number. After a successful upload request, the parts no longer exist and S3 response includes an **ETag** which uniquely identifies the combined object data. 
+
+### Configuration Parameters
+-----------------------------
+
+-   ***accessKey*** -   String
+    -   Mandatory Parameter    
+    -   Specifies the AWS access key to access Amazon S3 and has permissions to access the specified bucket.
+    -   Example value = AKIAJVAGFANC2LSZCJ4Q
+
+-   ***secretAccessKey***   -   String
+    -   Mandatory Parameter
+    -   Specifies the AWS secret access key to access Amazon S3 and has permissions to access the specified bucket.
+    -   Example value = wpVr3U82RmCKJoY007YfkaawT7CenhTcK1B8clue
+    
+-   ***endPoint***  -   String
+    -   Endpoint is the url that is entry point for a web service. Specify the valid endpoint to access S3 bucket.
+    -   Example value = s3.amazonaws.com
+
+-   ***bucketName***    -   String
+    -   Mandatory Parameter
+    -   S3 buckets are used to store objects which consists of data and metadata that describes the data. Specify the name of the bucket.
+    -   Example value = apex.app.test.s3
+
+-   ***outptuDirectoryPath***   -   String
+    -   Mandatory Parameter
+    -   Specifies the path of the output directory. 
+    -   Example value = dt/app/output
+
+-   ***mergerCount***   -   int
+    -   Specify the number of instances of S3FileMerger operator.
+    -   Default value = 1
+    
+-   ***timeOutWIndowCount***    -   int
+    -   Specify the count as streaming windows. This value will set to the operators in this module because the operators in this module is mostly interacts with the Amazon S3. Due to this reason, window id of these operators might be lag behind with the upstream operators.
```

Comment:
The intent of both questions is that the answer should be included in the document. Looks like this property maps to OperatorContext.TIMEOUT_WINDOW_COUNT; if so, that should be made clear.

Suggest something like this:

This property maps to the (OperatorContext.TIMEOUT_WINDOW_COUNT)[https://github.com/apache/apex-core/blob/master/api/src/main/java/com/datatorrent/api/Context.java) attribute and is a count of streaming windows. If specified, it will be set on all the operators of this module. Since these operators interact with S3, there may be additional latencies that cause the platform to kill them because they are considered stalled. Increasing this value prevents this and allows the application to proceed despite the latencies.